sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheckTest.java

```diff
@@ -0,0 +1,714 @@
+                primitiveType.set(isPrimitiveType());
+            }
+        }.check();
+        Assert.assertEquals(true, executed.get());
+        Assert.assertEquals(false, primitiveType.get());
+    }
+
+    @Test
+    public void testError() {
+        final AtomicBoolean executed = new AtomicBoolean(false);
+        final Jsr305AnnotationsCheck check = new Jsr305AnnotationsCheck();
+        final DetailAST ast = new DetailAST();
+        check.new AbstractJsr305Check(ast) {
+            @Override
+            protected void runcheck() {
+                executed.set(true);
+                Assert.assertEquals(false, isErrorFound());
+                error("foo");
+                Assert.assertEquals(true, isErrorFound());
+                error("foo");
+                Assert.assertEquals(true, isErrorFound());
+            }
+
+            @Override
+            protected void reportError(final String msg) {
+                // empty
+            }
+        }.check();
+        Assert.assertEquals(true, executed.get());
+    }
+
+    @Test
+    public void testContainsAll() {
+        final AtomicBoolean executed = new AtomicBoolean(false);
+        final AtomicBoolean containsAll = new AtomicBoolean(false);
+        final Jsr305AnnotationsCheck check = new Jsr305AnnotationsCheck();
+        final DetailAST ast = new DetailAST();
+        check.new AbstractJsr305Check(ast) {
+            @Override
+            protected void runcheck() {
+                executed.set(true);
+                containsAll.set(containsAll());
+            }
+        }.check();
+        Assert.assertEquals(true, executed.get());
+        Assert.assertEquals(true, containsAll.get());
+
+        executed.set(false);
+
+        check.new AbstractJsr305Check(ast) {
+            @Override
+            protected void runcheck() {
+                executed.set(true);
+                containsAll.set(containsAll(NullnessAnnotation.CHECK_FOR_NULL));
+            }
+        }.check();
+        Assert.assertEquals(true, executed.get());
+        Assert.assertEquals(false, containsAll.get());
+    }
+
+    @Test
+    public void testCollectLookedForAnnotations() {
+        final AtomicBoolean executed = new AtomicBoolean(false);
+        final Jsr305AnnotationsCheck check = new Jsr305AnnotationsCheck();
+        final Set<NullnessAnnotation> nullnessAnnotations = new HashSet<>();
+        final DetailAST ast = createAstWithTwoAnnotations();
+        final NullnessAnnotation[] annotationsToLookFor1 = new NullnessAnnotation[] {
+            NullnessAnnotation.CHECK_FOR_NULL, NullnessAnnotation.NONNULL,
+        };
+        check.new AbstractJsr305Check(ast) {
+            @Override
+            protected void runcheck() {
+                executed.set(true);
+                nullnessAnnotations.addAll(collectLookedForAnnotations(ast, annotationsToLookFor1));
+            }
+        }.check();
+        Assert.assertEquals(true, executed.get());
```

Comment:
I am suprised PMD allowed this. It was asking for a description to be added to all assertions.
Example: `Assert.assertTrue("XXX should be turned on", executed.get());`

Before fixing these, I would go back to https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/742#discussion_r275360370 as we do not prefer tests to be written like this. These tests look complicated and not clear what is going on.
I would go through each one and try to change it to a normal test or explain why you can't.

Reply:
See above, I have removed them.